### PR TITLE
Home work 14

### PR DESCRIPTION
**Pull request by AleksandrPitelin**
HomeWork_14/HW14.py

### CODE REVIEW

File: HomeWork_14/HW14.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    elif command == 'stats':
+        print(len(phonebook))
+    elif command == 'show':
+        s = input("Enter a name: ")
+        if phonebook.get(s):
+            print(phonebook[s])
+    elif command == 'add':
+        n = input("Enter a name: ")
+        p = int(input("Enter a phone: "))
+        tel = re.search(r"(\+?3?8?0)(\d+)", p)
```
Comment (AleksandrPitelin) [inline HomeWork_14/HW14.py:27]: отдельно работает, а в коде не хочет

File: HomeWork_14/HW14.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import re
+import json
+from datetime import datetime
+
+
+try:
+    with open('/Json_hw14', 'r') as file:
```
Comment (klymenok) [inline HomeWork_14/HW14.py:7]: `/` at the beginning of the filename tells that it is an absolute path. It means that the file will be created in the root of a filesystem which is not good. Remove that slash ('/')

File: HomeWork_14/HW14.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            else:
+                print('Name already exists')
+        else:
+            print('Invalid phone number')
+
+    elif command == 'delete':
+        n = input("Enter a name: ")
+        if phonebook.get(n):
+            del phonebook[n]
+            with open("/Json_hw14", 'w+') as file_json:
```
Comment (klymenok) [inline HomeWork_14/HW14.py:44]: `/` at the beginning of the filename tells that it is an absolute path. It means, that the file will be created in the root of a filesystem which is not good. Remove that slash ('/')

File: HomeWork_14/HW14.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    elif command == 'stats':
+        print(len(phonebook))
+    elif command == 'show':
+        s = input("Enter a name: ")
+        if phonebook.get(s):
+            print(phonebook[s])
+    elif command == 'add':
+        n = input("Enter a name: ")
+        p = int(input("Enter a phone: "))
+        tel = re.search(r"^(\\+380|380|0)[0-9]{9}$", p)
```
Comment (klymenok) [inline HomeWork_14/HW14.py:27]: It doesn't work because of `\\+`. Change it to `\+` and it will be correct

**Review by klymenok — CHANGES_REQUESTED** [review]

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**